### DISCUSSION TITLE
How to cancel change in textbox?

### DISCUSSION THREAD

**Discussion by bstrother1** [unoplatform/uno#9045] [Q&A]
What is the best way to cancel input in a Uno textbox with x:Bind before it reaches the backend of a two way binding so that it reverts to prior value?

**Comment by jeromelaban** [comment]
You can generally hook onto the `UIElement.LosingFocus` event and modify the `Text` property at that time, as by default a binding on `TextBox.Text` uses `UpdateSourceTrigger=LostFocus`.

That being said, if you're looking for validation you can use packages like those https://github.com/MADE-Apps/MADE.NET-UI, which [provide a validator](https://github.com/MADE-Apps/MADE.NET-UI/blob/main/docs/articles/features/ui-controls-validator.md).

**Reply by bstrother1** [reply]
Is DependencyProperty.UnsetValue not supported by Maui/Uno for x:Bind?

**Reply by jeromelaban** [reply]
What do you want to do with `DependencyProperty.UnsetValue` ?

**Reply by bstrother1** [reply]
Can't you use  DependencyProperty.UnsetValue to undo an invalid textbox change?

**Reply by jeromelaban** [reply]
Where did you find this feature? Is it something you used in WPF ?

**Reply by bstrother1** [reply]
https://docs.microsoft.com/en-us/dotnet/api/system.windows.dependencyproperty.unsetvalue?view=windowsdesktop-6.0

**Reply by jeromelaban** [reply]
Ok. Did you try with WinUI on Windows? It may be that this specific behavior is not supported.

**Reply by bstrother1** [reply]
It should also be supported by UWP and WPF https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.dependencyproperty.unsetvalue?view=winrt-22621   this is hosted under UWP documentation and I found this https://github.com/tom-englert/TomsToolbox/issues/4 which is a WPF package that mentions this.

**Reply by jeromelaban** [reply]
Can you make a sample that shows it working?
